include/superior_mysqlpp/utils.hpp

```diff
@@ -75,4 +75,224 @@ namespace SuperiorMySqlpp
     {
         return streamify(streamify(stream, arg), std::forward<Args>(args)...);
     }
+
+    namespace detail
+    {
+        /**
+         * Function information flags
+         */
+        enum : uint64_t
+        {
+            FunctionFlag_IsConst = 1ULL << 0ULL,
+            FunctionFlag_IsVolatile = 1ULL << 1ULL,
+            FunctionFlag_IsLvalueRefCtx = 1ULL << 3ULL,
+            FunctionFlag_IsRvalueRefCtx = 1ULL << 4ULL
+        };
+
+        /**
+         * @brief Generic function info obtained from FunctionInfo trait
+         * 
+         * @tparam ResultType type returned by function
+         * @tparam IsConst true_type, if function has been marked const
+         * @tparam IsVolatile true_type, if function has been marked volatile
+         * @tparam Args function arguments
+         */
+        template<typename ResultType, uint64_t Flags, typename... Args>
+        struct FunctionInfoImpl
+        {
+            using result_type   = ResultType;
+            using is_const      = std::integral_constant<bool, (Flags & FunctionFlag_IsConst) != 0>;
```

Comment:
May I ask why is this preferred over `static constexpr bool`?

Reply:
Because this is rather a trait, so you can easily use it as e.g. overload resolver (having 2 function overloads with one expecting false_type and the other true_type). And value extraction is very simple.

Reply:
I see it's mostly matter of preference then. Fair enough, I'm ok with this solution.